homeassistant/components/nfandroidtv/config_flow.py

```diff
@@ -2,22 +2,54 @@
 from __future__ import annotations
 
 import logging
+from typing import Any
 
 from notifications_android_tv.notifications import ConnectError, Notifications
 import voluptuous as vol
 
 from homeassistant import config_entries
+from homeassistant.components.dhcp import HOSTNAME, IP_ADDRESS, MAC_ADDRESS
 from homeassistant.const import CONF_HOST, CONF_NAME
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
 
-from .const import DEFAULT_NAME, DOMAIN
+from .const import ANDROID_TV_NAME, DEFAULT_NAME, DOMAIN, FIRE_TV_NAME
 
 _LOGGER = logging.getLogger(__name__)
 
 
 class NFAndroidTVFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for NFAndroidTV."""
 
+    def __init__(self):
+        """Initialize an NFAndroidTV flow."""
+        self.ip_address = None
+
+    async def async_step_dhcp(self, discovery_info) -> FlowResult:
+        """Handle dhcp discovery."""
+        self.ip_address = discovery_info[IP_ADDRESS]
+        mac = format_mac(discovery_info[MAC_ADDRESS])
+
+        for entry in self._async_current_entries():
+            # The mac is not always the correct format. Discovery seems to run twice.
+            # Once with 17 characters, once with missing 0s.
+            if entry.domain == DOMAIN and len(mac) == 17:
+                entry_updates = {**entry.data}
+                entry_updates[CONF_HOST] = self.ip_address
+                self.hass.config_entries.async_update_entry(
+                    entry, unique_id=mac, data=entry_updates
+                )
+                return self.async_abort(reason="already_configured")
+        await self.async_set_unique_id(mac)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
+        self._async_abort_entries_match({CONF_HOST: self.ip_address})
+        error = await self._async_try_connect(self.ip_address)
+        if error is None and len(mac) >= 12:
```

Comment:
For some reason, dhcp is running mac addresses twice. Once with the full 17 characters, once where 0s are actually missing, leaving a different unique id based in the mac address. This occurs whenever a mac has consecutive 0s.